new-explorer-frontend/src/components/App/App.js

```diff
@@ -1,27 +1,39 @@
-import { Route, Switch, Redirect, useHistory } from 'react-router';
+import { Route, Switch, Redirect, useHistory } from 'react-router-dom';
 import { useState, useEffect } from 'react';
 
 import Header from '../Header/Header';
 import Main from '../Main/Main';
-import SavedArticles from '../SavedArticles/SavedArticles';
+import SavedArticlesPage from '../SavedArticlesPage/SavedArticlesPage';
 import SearchSection from '../SearchSEction/SearchSection';
 import Footer from '../Footer/Footer';
 import SignInPopup from '../SignInPopup/SignInPopup';
 import SignUpPopup from '../SignUpPopup/SignUpPopup';
 import MessagePopup from '../MessagePopup/MessagePopup';
 import NotFound from '../NotFound/NotFound';
 import MenuPopup from '../MenuPopup/MenuPopup';
-import Preloader from '../Preloader/Preloader';
+import { CurrentUserContext } from '../../context/CurrentUserContext/CurrentUserContext';
+import SearchErrorMessage from '../SearchErrorMessage/SearchErrorMessage';
+
+import auth from '../../utils/auth';
+import newsApi from '../../utils/NewsApi';
+import mainApi from '../../utils/MainApi';
 
 import '../../index.css';
 
 function App() {
+  const history = useHistory();
+
+  const [userData, setUserData] = useState({});
   const [isSignInPopupOpen, setIsSignInPopupOpen] = useState(false);
   const [isSignUpPopupOpen, setIsSignUpPopupOpen] = useState(false);
   const [isMessageOpen, setIsMessageOpen] = useState(false);
   const [isMenuPopupOpen, setIsMenuPopupOpen] = useState(false);
   const [isNotFoundOpen, setIsNotFoundOpen] = useState(false);
   const [Isloding, setIsLoding] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+  const [savedArticles, setSavedArticles] = useState([]);
+  const [isSearchErrorOpen, setIsSearchErrorOpen] = useState(true);
 
   useEffect(() => {
     const close = (e) => {
```

Comment:
You shouldn't use `numbers` for checking what key has been pressed because such numbers don't tell other developers what kind of key it is. `(evt.key === “Escape”)`